hal/shared/spi_lock.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2020 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "spi_hal.h"
+
+namespace particle {
+
+class SpiConfigurationLock {
+public:
+    SpiConfigurationLock(hal_spi_interface_t spi)
```

Comment:
This is to avoid the need to have the configuration readily available at the time of the construction (this is problematic if we make `SpiConfigurationLock` a member variable). Since this is an internal class, we can perhaps add error reporting on `lock()` where this would be caught, or perhaps add an assertion there that it's properly configured.

Reply:
Added `SPARK_ASSERT`.